ibis/expr/scope.py

```diff
@@ -54,6 +54,27 @@ def __contains__(self, op):
     def __iter__(self):
         return iter(self._items.keys())
 
+    def set_value(
+        self, op: Node, timecontext: Optional[TimeContext], value: Any
+    ):
+        """ Given a op, timecontext and value, set op and (value, timecontext)
+            in scope.
+        Parameters
+        ----------
+        scope : collections.Mapping
+            a dictionary mapping :class:`~ibis.expr.operations.Node`
+            subclass instances to concrete data, and the time context associate
+            with it (if any).
+        op: ibis.expr.operations.Node
+            key in scope.
+        timecontext: Optional[TimeContext]
+        value: Any
+            the cached result to save in scope, an object whose type may
+            differ in different backends.
+        """
+        if self.get_value(op, timecontext) is None:
```

Comment:
It is a little weird for a "set" method to do nothing if there is a value associated with the key. Why are we doing that?

Reply:
If there is a value associated with the key, but time context is smaller than the current time context we are going to set, `get_value` will return `None` and we will proceed to set the new value in scope.
The API takes `timecontext` as a param, I think it suggests we will set according to the time context. Should we change the name of this function, or add a more specific doc here to avoid confusion?

Reply:
I see. This is fine. Please explain the behavior in the docstring and add tests.

Reply:
Sure, explained in docstring. Will add tests in followups

Reply:
yep, can you add a comment to this effect